api/src/models/user/user.schema.ts

```diff
@@ -0,0 +1,33 @@
+import { Schema } from "mongoose";
+
+const UserSchema: Schema = new Schema({
+  firstName: String,
+  lastName: String,
+  auth: {
+    email: {
+      type: String,
+      required: false
+    },
+    password: {
+      type: String,
+      required: false,
+    },
+    oauth: {
+      type: String,
+      required: false
+    }
+  },
+  avatar: {
+    type: [{url: String}],
+  },
+  connections: {},
+  connectionOf: {},
+  threads: {
+    started:{},
+    commented: {},
+    liked: {},
+    shared:{}
```

Comment:
I think there is a better way to type these if we decide to implement this feature.

Reply:
Let come back to look at this after the prototype. We could look over our user actions flow and work out what would be best.